### PR TITLE
multi-connection support for NH, DocDB, EasyTables; updating extensio…

### PR DESCRIPTION
**Pull request by brettsam**
…n Nugets

### CODE REVIEW

File: src/WebJobs.Script/Binding/DocumentDBBinding.cs [hunk 1]
```diff
@@ -33,6 +34,8 @@ internal class DocumentDBBinding : FunctionBinding
 
         public bool CreateIfNotExists { get; private set; }
 
+        public string ConnectionString { get; private set; }
```
Comment (mathewc) [inline src/WebJobs.Script/Binding/DocumentDBBinding.cs]: I think your actual attribute uses the name "ConnectionString" even though it is really an app setting name, not a connection string. I know the xml doc explains things, but we should probably rename these consistently to something like "Connection".

File: src/WebJobs.Script/Binding/EasyTableBinding.cs [hunk 2]
```diff
@@ -22,6 +22,9 @@ public class EasyTableBinding : FunctionBinding
         {
             TableName = metadata.TableName;
             Id = metadata.Id;
+            MobileAppUri = metadata.Connection;
```
Comment (mathewc) [inline src/WebJobs.Script/Binding/EasyTableBinding.cs]: Yeah, I think my naming worked out :) "Connection" does make sense for a URI to connect to as well :)

File: src/WebJobs.Script/Binding/NotificationHubBinding.cs [hunk 3]
```diff
@@ -21,11 +21,18 @@ internal class NotificationHubBinding : FunctionBinding
             base(config, metadata, access)
         {
             TagExpression = metadata.TagExpression;
+            ConnectionString = metadata.Connection;
```
Comment (mathewc) [inline src/WebJobs.Script/Binding/NotificationHubBinding.cs]: This ConnectionString / Connection interpretation is going to cause confusion for us until we standardize things

File: src/WebJobs.Script/Binding/EasyTableBinding.cs [hunk 4]
```diff
@@ -29,6 +32,10 @@ public class EasyTableBinding : FunctionBinding
 
         public string Id { get; private set; }
 
+        public string MobileAppUri { get; private set; }
+
+        public string ApiKey { get; private set; }
+
```
Comment (brettsam) [inline src/WebJobs.Script/Binding/EasyTableBinding.cs]: @mathewc -- is there a reason that these all of these bindings need to have public properties like this? I was tempted to remove them and just store the metadata as a private field. But I wasn't sure if something somewhere consumed them...

**Comment by mathewc** [comment]
Looks good. :ship:

**Pull request closed without merging**